diff --git a/chrome/browser/ui/webui/chrome_untrusted_web_ui_configs.cc b/chrome/browser/ui/webui/chrome_untrusted_web_ui_configs.cc
index d14048efd86341cec53e7f438380252dde7f5523..84705abd9050a61284dc2024645b555fd6c54188 100644
--- a/chrome/browser/ui/webui/chrome_untrusted_web_ui_configs.cc
+++ b/chrome/browser/ui/webui/chrome_untrusted_web_ui_configs.cc
@@ -4,11 +4,55 @@

 #include "chrome/browser/ui/webui/chrome_untrusted_web_ui_configs.h"

+#include "base/feature_list.h"
+#include "brave/components/ai_chat/core/common/buildflags/buildflags.h"
+#include "brave/components/brave_news/common/buildflags/buildflags.h"
+#include "brave/components/brave_vpn/common/buildflags/buildflags.h"
+#include "brave/components/brave_wallet/common/buildflags/buildflags.h"
+#include "brave/components/local_ai/buildflags/buildflags.h"
+#include "brave/components/playlist/core/common/buildflags/buildflags.h"
 #include "build/build_config.h"
 #include "chrome/common/buildflags.h"
+#include "components/history_embeddings/core/history_embeddings_features.h"
 #include "content/public/browser/webui_config_map.h"
 #include "printing/buildflags/buildflags.h"

+#if BUILDFLAG(ENABLE_BRAVE_WALLET)
+#include "brave/browser/ui/webui/brave_wallet/ledger/ledger_ui.h"
+#include "brave/browser/ui/webui/brave_wallet/line_chart/line_chart_ui.h"
+#include "brave/browser/ui/webui/brave_wallet/market/market_ui.h"
+#include "brave/browser/ui/webui/brave_wallet/nft/nft_ui.h"
+#include "brave/browser/ui/webui/brave_wallet/trezor/trezor_ui.h"
+#endif
+
+#if BUILDFLAG(ENABLE_AI_CHAT)
+#include "brave/browser/ui/webui/ai_chat/ai_chat_untrusted_conversation_ui.h"
+#include "brave/browser/ui/webui/ai_chat/chart_display_ui.h"
+#include "brave/browser/ui/webui/ai_chat/code_sandbox_ui.h"
+#include "brave/components/ai_chat/core/common/features.h"
+#endif
+
+#if BUILDFLAG(ENABLE_BRAVE_NEWS) && !BUILDFLAG(IS_ANDROID)
+#include "brave/browser/ui/webui/brave_news/brave_news_ui.h"
+#include "brave/components/brave_news/common/features.h"
+#endif
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+#if !BUILDFLAG(IS_ANDROID)
+#include "brave/browser/ui/webui/brave_vpn/vpn_panel_ui.h"
+#endif
+#include "brave/components/brave_vpn/common/brave_vpn_utils.h"
+#endif
+
+#if BUILDFLAG(ENABLE_PLAYLIST_WEBUI)
+#include "brave/browser/ui/webui/playlist_ui.h"
+#include "brave/components/playlist/core/common/features.h"
+#endif
+
+#if BUILDFLAG(ENABLE_LOCAL_AI)
+#include "brave/browser/ui/webui/local_ai/local_ai_ui.h"
+#endif
+
 #if defined(TOOLKIT_VIEWS)
 #include "chrome/browser/ui/webui/data_sharing/data_sharing_ui.h"
 #include "chrome/browser/ui/webui/side_panel/read_anything/read_anything_untrusted_ui.h"
@@ -42,9 +86,14 @@ void RegisterChromeUntrustedWebUIConfigs() {
   ash::RegisterAshChromeUntrustedWebUIConfigs();
 #endif  // BUILDFLAG(IS_CHROMEOS)

-#if defined(TOOLKIT_VIEWS) || BUILDFLAG(ENABLE_PRINT_PREVIEW)
+#if defined(TOOLKIT_VIEWS) || BUILDFLAG(ENABLE_PRINT_PREVIEW) || \
+    BUILDFLAG(ENABLE_BRAVE_WALLET) || BUILDFLAG(ENABLE_LOCAL_AI) || \
+    BUILDFLAG(ENABLE_AI_CHAT) || \
+    (BUILDFLAG(ENABLE_BRAVE_VPN) && !BUILDFLAG(IS_ANDROID)) || \
+    (BUILDFLAG(ENABLE_PLAYLIST_WEBUI) && !BUILDFLAG(IS_ANDROID)) || \
+    (BUILDFLAG(ENABLE_BRAVE_NEWS) && !BUILDFLAG(IS_ANDROID))
   auto& map = content::WebUIConfigMap::GetInstance();
-#endif  // defined(TOOLKIT_VIEWS) || BUILDFLAG(ENABLE_PRINT_PREVIEW)
+#endif

 #if defined(TOOLKIT_VIEWS)
   map.AddUntrustedWebUIConfig(
@@ -77,4 +126,60 @@ void RegisterChromeUntrustedWebUIConfigs() {
   map.AddUntrustedWebUIConfig(
       std::make_unique<NtpMicrosoftAuthUntrustedUIConfig>());
 #endif  // !BUILDFLAG(IS_ANDROID)
+
+#if BUILDFLAG(ENABLE_BRAVE_WALLET)
+  map.AddUntrustedWebUIConfig(
+      std::make_unique<market::UntrustedMarketUIConfig>());
+  map.AddUntrustedWebUIConfig(
+      std::make_unique<nft::UntrustedNftUIConfig>());
+  map.AddUntrustedWebUIConfig(
+      std::make_unique<line_chart::UntrustedLineChartUIConfig>());
+#if !BUILDFLAG(IS_ANDROID)
+  map.AddUntrustedWebUIConfig(
+      std::make_unique<ledger::UntrustedLedgerUIConfig>());
+  map.AddUntrustedWebUIConfig(
+      std::make_unique<trezor::UntrustedTrezorUIConfig>());
+#endif
+#endif
+
+#if BUILDFLAG(ENABLE_LOCAL_AI)
+  if (base::FeatureList::IsEnabled(history_embeddings::kHistoryEmbeddings)) {
+    map.AddUntrustedWebUIConfig(
+        std::make_unique<local_ai::UntrustedLocalAIUIConfig>());
+  }
+#endif
+
+#if !BUILDFLAG(IS_ANDROID)
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  if (brave_vpn::IsBraveVPNFeatureEnabled()) {
+    map.AddUntrustedWebUIConfig(std::make_unique<UntrustedVPNPanelUIConfig>());
+  }
+#endif
+
+#if BUILDFLAG(ENABLE_PLAYLIST_WEBUI)
+  if (base::FeatureList::IsEnabled(playlist::features::kPlaylist)) {
+    map.AddUntrustedWebUIConfig(
+        std::make_unique<playlist::UntrustedPlaylistUIConfig>());
+    map.AddUntrustedWebUIConfig(
+        std::make_unique<playlist::UntrustedPlaylistPlayerUIConfig>());
+  }
+#endif
+#endif
+
+#if BUILDFLAG(ENABLE_AI_CHAT)
+  if (ai_chat::features::IsAIChatEnabled()) {
+    map.AddUntrustedWebUIConfig(
+        std::make_unique<AIChatUntrustedConversationUIConfig>());
+    map.AddUntrustedWebUIConfig(
+        std::make_unique<ai_chat::ChartDisplayUIConfig>());
+    map.AddUntrustedWebUIConfig(
+        std::make_unique<ai_chat::CodeSandboxUIConfig>());
+  }
+#endif
+
+#if BUILDFLAG(ENABLE_BRAVE_NEWS) && !BUILDFLAG(IS_ANDROID)
+  if (base::FeatureList::IsEnabled(brave_news::features::kBraveNewsSidebar)) {
+    map.AddUntrustedWebUIConfig(std::make_unique<BraveNewsUIConfig>());
+  }
+#endif
 }
