diff --git a/components/sync/engine/BUILD.gn b/components/sync/engine/BUILD.gn
index 35733ce4eb0af66b93f935cc296fc55f815be3e9..5c54cf9cdcfc0a49e2793354d3cb996415f4b8eb 100644
--- a/components/sync/engine/BUILD.gn
+++ b/components/sync/engine/BUILD.gn
@@ -156,6 +156,9 @@ static_library("engine") {
     "traffic_logger.h",
     "update_handler.h",
   ]
+  import("//brave/components/sync/engine/sources.gni")
+  sources -= brave_components_sync_engine_excluded_sources
+  sources += brave_components_sync_engine_sources

   configs += [
     # The target has >50 .cc files, so speed up the build with
