diff --git a/components/sync/base/BUILD.gn b/components/sync/base/BUILD.gn
index 1a8f0297c7974fa85f1c38a6e75a30ea6f074539..25f9a81385e3cb641f7309657a8bc375f1bbd6c5 100644
--- a/components/sync/base/BUILD.gn
+++ b/components/sync/base/BUILD.gn
@@ -89,6 +89,10 @@ static_library("base") {
   ]

   defines = [ "SYNC_USER_AGENT_PRODUCT=$sync_user_agent_product" ]
+  import("//brave/components/sync/base/sources.gni")
+  sources -= brave_components_sync_base_excluded_sources
+  sources += brave_components_sync_base_sources
+  public_deps += brave_components_sync_base_public_deps
 }

 fuzzer_test("unique_position_fuzzer") {
