diff --git a/content/browser/web_contents/web_contents_impl.cc b/content/browser/web_contents/web_contents_impl.cc
index 7edad76e875c081e4e8d7de917658d006624de91..787cc149b958735ba44ad61c1c46f5d8f507ed88 100644
--- a/content/browser/web_contents/web_contents_impl.cc
+++ b/content/browser/web_contents/web_contents_impl.cc
@@ -1637,6 +1637,17 @@ std::vector<raw_ptr<WebContentsImpl>> WebContentsImpl::GetAllWebContents() {
   return result;
 }
 
+// static
+void WebContents::SyncRendererPrefsForBrowserContext(
+    BrowserContext* browser_context) {
+  DCHECK(browser_context);
+  for (WebContentsImpl* web_contents : WebContentsImpl::GetAllWebContents()) {
+    if (web_contents->GetBrowserContext() == browser_context) {
+      web_contents->SyncRendererPrefs();
+    }
+  }
+}
+
 // static
 WebContentsImpl* WebContentsImpl::FromFrameTreeNode(
     const FrameTreeNode* frame_tree_node) {
@@ -5560,6 +5571,7 @@ FrameTree* WebContentsImpl::CreateNewWindow(
   // SiteInstance is created in a new BrowsingInstance.
   scoped_refptr<SiteInstance> site_instance;
   if (params.opener_suppressed) {
+    BRAVE_WEB_CONTENTS_IMPL_CREATE_NEW_WINDOW_OPENER_SUPPRESSED
     if (is_guest) {
       // For guests, noopener windows can be created in a new BrowsingInstance
       // as long as they preserve the guest's StoragePartition.
@@ -5657,7 +5669,7 @@ FrameTree* WebContentsImpl::CreateNewWindow(
 
   // Save the window for later if we're not suppressing the opener (since it
   // will be shown immediately).
-  if (!params.opener_suppressed) {
+  if (!params.opener_suppressed && !params.no_javascript_access) {
     if (!is_guest) {
       WebContentsView* new_view = new_contents_impl->view_.get();
 
@@ -9676,6 +9688,8 @@ const blink::RendererPreferences& WebContentsImpl::GetRendererPrefs(
     return guest->GetRendererPrefs();
   }
   RenderViewHostImpl::GetPlatformSpecificPrefs(&renderer_preferences_);
+  GetContentClient()->browser()->UpdateRendererPreferences(
+      GetBrowserContext(), &renderer_preferences_);
   return renderer_preferences_;
 }
 
