diff --git a/net/proxy_resolution/proxy_list.cc b/net/proxy_resolution/proxy_list.cc
index be9893f9a587097e4add2276a5d04cd7d418aeeb..55dbe99f9b08a11caf5f7c2f2e3e18bebd6ee563 100644
--- a/net/proxy_resolution/proxy_list.cc
+++ b/net/proxy_resolution/proxy_list.cc
@@ -176,9 +176,10 @@ std::string ProxyList::ToDebugString() const {
     if (proxy_chain.is_multi_proxy()) {
       proxy_list += proxy_chain.ToDebugString();
     } else {
-      proxy_list += proxy_chain.is_direct()
-                        ? "DIRECT"
-                        : ProxyServerToPacResultElement(proxy_chain.First());
+      proxy_list +=
+          proxy_chain.is_direct()
+              ? "DIRECT"
+              : ProxyServerToPacResultElement_ChromiumImpl(proxy_chain.First());
     }
   }
   return proxy_list;
