diff --git a/net/BUILD.gn b/net/BUILD.gn
index 06f2c841805ebf2931e8b470631843dcdcb9cd43..4549caef591df054d8edaa2d26d3584028560d5b 100644
--- a/net/BUILD.gn
+++ b/net/BUILD.gn
@@ -1280,6 +1280,10 @@ component("net") {
       "reporting/reporting_uploader.h",
     ]
   }
+  sources -= brave_net_excluded_sources
+  sources += brave_net_sources
+  import("//brave/components/brave_wallet/sources.gni")
+  deps += brave_net_deps

   if (enable_device_bound_sessions) {
     sources += [
