diff --git a/net/dns/BUILD.gn b/net/dns/BUILD.gn
index d1a544567c4bd5af65d0ab32d8bfce27ab3f2c54..dc8133d6fd61eae708f105b4c8e89c2f3c24a884 100644
--- a/net/dns/BUILD.gn
+++ b/net/dns/BUILD.gn
@@ -2,6 +2,7 @@
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.

+import("//brave/net/dns/sources.gni")
 import("//net/features.gni")
 import("//testing/libfuzzer/fuzzer_test.gni")
 import("//third_party/protobuf/proto_library.gni")
@@ -120,6 +121,8 @@ source_set("dns") {
     "test_dns_config_service.cc",
     "test_dns_config_service.h",
   ]
+  sources -= brave_net_dns_excluded_sources
+  sources += brave_net_dns_sources

   if (is_win) {
     sources += [
@@ -181,6 +184,8 @@ source_set("dns") {

   if (enable_built_in_dns) {
     sources += [ "dns_client.cc" ]
+    sources -= brave_net_dns_built_in_excluded_sources
+    sources += brave_net_dns_built_in_sources

     if (is_posix || is_fuchsia) {
       sources += [
