diff --git a/services/network/BUILD.gn b/services/network/BUILD.gn
index 66c879178491c2e21962f12a190eda179ce7a2c3..837ecab282f9b1c4dbdf81813fd9cafdb3bb7e2b 100644
--- a/services/network/BUILD.gn
+++ b/services/network/BUILD.gn
@@ -397,6 +397,10 @@ component("network_service") {
     "//third_party/zstd:decompress",
     "//url",
   ]
+  import("//brave/services/network/sources.gni")
+  sources -= brave_services_network_excluded_sources
+  sources += brave_services_network_sources
+  deps += brave_services_network_deps
 
   if (is_linux || is_chromeos) {
     deps += [
