diff --git a/services/network/BUILD.gn b/services/network/BUILD.gn
index 66c879178491c2e21962f12a190eda179ce7a2c3..dddddb9878663ec6c200c398c4b612c5d443e549 100644
--- a/services/network/BUILD.gn
+++ b/services/network/BUILD.gn
@@ -397,6 +397,10 @@ component("network_service") {
     "//third_party/zstd:decompress",
     "//url",
   ]
+  import("//brave/services/network/sources.gni")
+  sources -= brave_services_network_excluded_sources
+  sources += brave_services_network_sources
+  deps += brave_services_network_deps

   if (is_linux || is_chromeos) {
     deps += [
