diff --git a/third_party/blink/renderer/modules/mediastream/webaudio_media_stream_audio_sink.h b/third_party/blink/renderer/modules/mediastream/webaudio_media_stream_audio_sink.h
index b993aff953298250a3fe395525dee25f9427191c..f0c498f68018e412ad504b081f8d24919847eb96 100644
--- a/third_party/blink/renderer/modules/mediastream/webaudio_media_stream_audio_sink.h
+++ b/third_party/blink/renderer/modules/mediastream/webaudio_media_stream_audio_sink.h
@@ -46,11 +46,10 @@ class MODULES_EXPORT WebAudioMediaStreamAudioSink
       public media::AudioConverter::InputCallback,
       public WebMediaStreamAudioSink {
  public:
-  static const int kWebAudioRenderBufferSize;
-
   WebAudioMediaStreamAudioSink(MediaStreamComponent* component,
                                int context_sample_rate,
-                               base::TimeDelta platform_buffer_duration);
+                               base::TimeDelta platform_buffer_duration,
+                               uint32_t render_quantum_frames);
 
   WebAudioMediaStreamAudioSink(const WebAudioMediaStreamAudioSink&) = delete;
   WebAudioMediaStreamAudioSink& operator=(const WebAudioMediaStreamAudioSink&) =
