diff --git a/third_party/blink/renderer/core/frame/navigator_language.cc b/third_party/blink/renderer/core/frame/navigator_language.cc
index 75cd3461317b7df6bcf666c24d8b14a1e631f2a1..796a9e0aee05dc89844a6b32b0a43134e0420c40 100644
--- a/third_party/blink/renderer/core/frame/navigator_language.cc
+++ b/third_party/blink/renderer/core/frame/navigator_language.cc
@@ -5,6 +5,7 @@
 #include "third_party/blink/renderer/core/frame/navigator_language.h"

 #include "base/command_line.h"
+#include "brave/third_party/blink/renderer/core/farbling/brave_session_cache.h"
 #include "services/network/public/cpp/features.h"
 #include "third_party/blink/public/common/switches.h"
 #include "third_party/blink/renderer/core/frame/web_feature.h"
@@ -84,6 +85,19 @@ void NavigatorLanguage::EnsureUpdatedLanguage() {
   }

   if (languages_dirty_) {
+    const BraveFarblingLevel farbling_level = brave::GetBraveFarblingLevelFor(
+        execution_context_, ContentSettingsType::BRAVE_WEBCOMPAT_LANGUAGE,
+        BraveFarblingLevel::OFF);
+    if (farbling_level != BraveFarblingLevel::OFF && execution_context_) {
+      if (auto persona_languages =
+              brave::BraveSessionCache::From(*execution_context_)
+                  .PersonaLanguages()) {
+        languages_ = *persona_languages;
+        languages_dirty_ = false;
+        return;
+      }
+    }
+
     languages_ = ParseAndSanitize(GetAcceptLanguages());
     // Reduce the Accept-Language if the ReduceAcceptLanguage deprecation
     // trial is not enabled and feature flag ReduceAcceptLanguage is enabled.
@@ -97,6 +111,11 @@ void NavigatorLanguage::EnsureUpdatedLanguage() {
                    blink::switches::kDisableReduceAcceptLanguage)) {
       languages_ = Vector<String>({languages_.front()});
     }
+    if (farbling_level == BraveFarblingLevel::MAXIMUM) {
+      languages_ = Vector<String>({"en-US", "en"});
+    } else if (farbling_level == BraveFarblingLevel::BALANCED) {
+      languages_.Shrink(1);
+    }
     languages_dirty_ = false;
   }
 }
