diff --git a/third_party/blink/renderer/modules/mediastream/media_stream_track.h b/third_party/blink/renderer/modules/mediastream/media_stream_track.h
index 807d07d2c78f3cdf8ccabe455b4042695665cab8..7b35774a377bbaa0f972233abc68b3bfea8b19f3 100644
--- a/third_party/blink/renderer/modules/mediastream/media_stream_track.h
+++ b/third_party/blink/renderer/modules/mediastream/media_stream_track.h
@@ -146,7 +146,8 @@ class MODULES_EXPORT MediaStreamTrack
 
   virtual std::unique_ptr<AudioSourceProvider> CreateWebAudioSource(
       int context_sample_rate,
-      base::TimeDelta platform_buffer_duration) = 0;
+      base::TimeDelta platform_buffer_duration,
+      uint32_t render_quantum_frames) = 0;
 
   virtual ImageCapture* GetImageCapture() = 0;
   virtual std::optional<const MediaStreamDevice> device() const = 0;
