diff --git a/third_party/blink/renderer/core/exported/web_view_impl.cc b/third_party/blink/renderer/core/exported/web_view_impl.cc
index 59f2f2f70904bdd6c691fbe03fdd84b57e748627..3264aefd3a8dbc1883009bce0898dfabdc8ffc18 100644
--- a/third_party/blink/renderer/core/exported/web_view_impl.cc
+++ b/third_party/blink/renderer/core/exported/web_view_impl.cc
@@ -168,6 +168,7 @@
 #include "third_party/blink/renderer/core/scroll/scrollbar_theme.h"
 #include "third_party/blink/renderer/core/skeleton/skeleton_loader.h"
 #include "third_party/blink/renderer/core/speculation_rules/document_speculation_rules.h"
+#include "third_party/blink/renderer/core/timezone/timezone_controller.h"
 #include "third_party/blink/renderer/core/timing/dom_window_performance.h"
 #include "third_party/blink/renderer/core/timing/performance.h"
 #include "third_party/blink/renderer/core/timing/window_performance.h"
@@ -3732,6 +3733,10 @@ void WebViewImpl::UpdateRendererPreferences(
     const RendererPreferences& preferences) {
   std::string old_accept_languages = renderer_preferences_.accept_languages;
   renderer_preferences_ = preferences;
+  if (renderer_preferences_.fingerprint_browser_timezone_override_initialized) {
+    TimeZoneController::SetFingerprintBrowserTimezoneOverride(String::FromUtf8(
+        renderer_preferences_.fingerprint_browser_timezone_override));
+  }

   for (auto& watcher : renderer_preference_watchers_) {
     watcher->NotifyUpdate(renderer_preferences_);
