diff --git a/third_party/blink/renderer/core/BUILD.gn b/third_party/blink/renderer/core/BUILD.gn
index e84a67cee87b9f77d894fbcc83f1f21df5973960..85b36d4b2b0c94d2a346132ee119aadf7f8a60a5 100644
--- a/third_party/blink/renderer/core/BUILD.gn
+++ b/third_party/blink/renderer/core/BUILD.gn
@@ -264,6 +264,16 @@ component("core") {
   ]

   sources += blink_core_sources_bindings
+  sources -= [
+    "//third_party/blink/renderer/bindings/core/v8/js_event_handler_for_content_attribute.cc",
+    "//third_party/blink/renderer/bindings/core/v8/referrer_script_info.cc",
+    "//third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc",
+  ]
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/bindings/core/v8/js_event_handler_for_content_attribute.cc",
+    "//brave/chromium_src/third_party/blink/renderer/bindings/core/v8/referrer_script_info.cc",
+    "//brave/chromium_src/third_party/blink/renderer/bindings/core/v8/v8_script_runner.cc",
+  ]

   # If you create a new subdirectory 'foo', make 'foo/build.gni' to define
   # blink_core_sources_foo, and include it here. Add any dependencies in the
@@ -278,17 +288,47 @@ component("core") {
                          "",
                          "canvas_interventions")
   sources += rebase_path(blink_core_sources_clipboard, "", "clipboard")
+  sources -= rebase_path([ "system_clipboard.cc" ], "", "clipboard")
+  sources += [ "//brave/chromium_src/third_party/blink/renderer/core/clipboard/system_clipboard.cc" ]
   sources +=
       rebase_path(blink_core_sources_content_capture, "", "content_capture")
   sources +=
       rebase_path(blink_core_sources_context_features, "", "context_features")
   sources += rebase_path(blink_core_sources_css, "", "css")
+  sources -= rebase_path([
+                           "media_values.cc",
+                           "media_values_cached.cc",
+                           "media_values_dynamic.cc",
+                         ],
+                         "",
+                         "css")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/css/media_values.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/css/media_values_cached.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/css/media_values_dynamic.cc",
+  ]
   sources += rebase_path(blink_core_sources_display_lock, "", "display_lock")
   sources +=
       rebase_path(blink_core_sources_view_transition, "", "view_transition")
   sources += rebase_path(blink_core_sources_script_tools, "", "script_tools")
   sources += rebase_path(blink_core_sources_dom, "", "dom")
+  sources -= rebase_path([
+                           "events/event_listener_map.cc",
+                           "events/event_target.cc",
+                           "events/registered_event_listener.cc",
+                           "node.cc",
+                         ],
+                         "",
+                         "dom")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/dom/events/event_listener_map.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/dom/events/event_target.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/dom/events/registered_event_listener.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/dom/node.cc",
+  ]
   sources += rebase_path(blink_core_sources_editing, "", "editing")
+  sources -= rebase_path([ "commands/clipboard_commands.cc" ], "", "editing")
+  sources += [ "//brave/chromium_src/third_party/blink/renderer/core/editing/commands/clipboard_commands.cc" ]
   sources += rebase_path(blink_core_sources_events, "", "events")
   sources +=
       rebase_path(blink_core_sources_execution_context, "", "execution_context")
@@ -301,6 +341,34 @@ component("core") {
                          "",
                          "fragment_directive")
   sources += rebase_path(blink_core_sources_frame, "", "frame")
+  sources -= rebase_path([
+                           "cached_permission_status.cc",
+                           "dom_window.cc",
+                           "local_dom_window.cc",
+                           "local_frame.cc",
+                           "local_frame_mojo_handler.cc",
+                           "location.cc",
+                           "navigator_device_memory.cc",
+                           "navigator_language.cc",
+                           "reporting_observer.cc",
+                           "screen.cc",
+                           "web_local_frame_impl.cc",
+                         ],
+                         "",
+                         "frame")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/cached_permission_status.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/dom_window.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/local_dom_window.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/local_frame.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/local_frame_mojo_handler.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/location.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/navigator_device_memory.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/navigator_language.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/reporting_observer.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/screen.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/frame/web_local_frame_impl.cc",
+  ]
   sources += rebase_path(blink_core_sources_ad_tracker, "", "ad_tracker")
   sources += get_target_outputs(":make_deprecation_info")
   sources += rebase_path(blink_core_sources_fullscreen, "", "fullscreen")
@@ -308,11 +376,33 @@ component("core") {
   sources += rebase_path(blink_core_sources_geometry, "", "geometry")
   sources += rebase_path(blink_core_sources_highlight, "", "highlight")
   sources += rebase_path(blink_core_sources_html, "", "html")
+  sources -= rebase_path([
+                           "canvas/canvas_async_blob_creator.cc",
+                           "canvas/html_canvas_element.cc",
+                           "html_link_element.cc",
+                           "html_meta_element.cc",
+                           "html_script_element.cc",
+                           "media/autoplay_policy.cc",
+                           "media/html_media_element.cc",
+                         ],
+                         "",
+                         "html")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/html/canvas/canvas_async_blob_creator.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/html/canvas/html_canvas_element.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/html/html_link_element.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/html/html_meta_element.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/html/html_script_element.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/html/media/autoplay_policy.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/html/media/html_media_element.cc",
+  ]
   sources +=
       rebase_path(blink_core_sources_image_replacement, "", "image_replacement")
   sources += rebase_path(blink_core_sources_imagebitmap, "", "imagebitmap")
   sources += rebase_path(blink_core_sources_input, "", "input")
   sources += rebase_path(blink_core_sources_inspector, "", "inspector")
+  sources -= rebase_path([ "inspector_page_agent.cc" ], "", "inspector")
+  sources += [ "//brave/chromium_src/third_party/blink/renderer/core/inspector/inspector_page_agent.cc" ]
   sources += rebase_path(blink_core_sources_intersection_observer,
                          "",
                          "intersection_observer")
@@ -321,6 +411,38 @@ component("core") {
                          "",
                          "lcp_critical_path_predictor")
   sources += rebase_path(blink_core_sources_loader, "", "loader")
+  sources -= rebase_path([
+                           "anchor_element_interaction_tracker.cc",
+                           "base_fetch_context.cc",
+                           "cookie_jar.cc",
+                           "document_load_timing.cc",
+                           "document_loader.cc",
+                           "frame_fetch_context.cc",
+                           "image_loader.cc",
+                           "link_loader.cc",
+                           "mixed_content_checker.cc",
+                           "modulescript/module_script_loader.cc",
+                           "modulescript/module_tree_linker.cc",
+                           "resource/svg_document_resource.cc",
+                           "worker_fetch_context.cc",
+                         ],
+                         "",
+                         "loader")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/anchor_element_interaction_tracker.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/base_fetch_context.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/cookie_jar.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/document_load_timing.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/document_loader.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/frame_fetch_context.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/image_loader.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/link_loader.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/mixed_content_checker.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/modulescript/module_script_loader.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/modulescript/module_tree_linker.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/resource/svg_document_resource.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/loader/worker_fetch_context.cc",
+  ]
   sources += rebase_path(blink_core_sources_mathml, "", "mathml")
   sources += rebase_path(blink_core_sources_messaging, "", "messaging")
   sources +=
@@ -333,6 +455,20 @@ component("core") {
   sources += rebase_path(blink_core_sources_origin_trials, "", "origin_trials")
   sources += rebase_path(blink_core_sources_overscroll, "", "overscroll")
   sources += rebase_path(blink_core_sources_page, "", "page")
+  sources -= rebase_path([
+                           "chrome_client.cc",
+                           "chrome_client_impl.cc",
+                           "create_window.cc",
+                           "frame_tree.cc",
+                         ],
+                         "",
+                         "page")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/page/chrome_client.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/page/chrome_client_impl.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/page/create_window.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/page/frame_tree.cc",
+  ]
   sources += rebase_path(blink_core_sources_paint, "", "paint")
   sources +=
       rebase_path(blink_core_sources_resize_observer, "", "resize_observer")
@@ -342,7 +478,21 @@ component("core") {
   sources +=
       rebase_path(blink_core_sources_speculation_rules, "", "speculation_rules")
   sources += rebase_path(blink_core_sources_scheduler, "", "scheduler")
+  sources -= rebase_path([ "scheduled_action.cc" ], "", "scheduler")
+  sources += [ "//brave/chromium_src/third_party/blink/renderer/core/scheduler/scheduled_action.cc" ]
   sources += rebase_path(blink_core_sources_script, "", "script")
+  sources -= rebase_path([
+                           "detect_javascript_frameworks.cc",
+                           "dynamic_module_resolver.cc",
+                           "script_loader.cc",
+                         ],
+                         "",
+                         "script")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/script/detect_javascript_frameworks.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/script/dynamic_module_resolver.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/script/script_loader.cc",
+  ]
   sources += rebase_path(blink_core_sources_scroll, "", "scroll")
   sources += rebase_path(blink_core_sources_shadow_realm, "", "shadow_realm")
   sources += rebase_path(blink_core_sources_streams, "", "streams")
@@ -350,10 +500,15 @@ component("core") {
   sources += rebase_path(blink_core_sources_svg, "", "svg")
   sources += rebase_path(blink_core_sources_timezone, "", "timezone")
   sources += rebase_path(blink_core_sources_timing, "", "timing")
+  sources -= rebase_path([ "time_clamper.cc" ], "", "timing")
+  sources += [ "//brave/chromium_src/third_party/blink/renderer/core/timing/time_clamper.cc" ]
   sources += rebase_path(blink_core_sources_trustedtypes, "", "trustedtypes")
   sources += rebase_path(blink_core_sources_typed_arrays, "", "typed_arrays")
   sources += rebase_path(blink_core_sources_url, "", "url")
   sources += rebase_path(blink_core_sources_workers, "", "workers")
+  sources -=
+      rebase_path([ "shared_worker_content_settings_proxy.cc" ], "", "workers")
+  sources += [ "//brave/chromium_src/third_party/blink/renderer/core/workers/shared_worker_content_settings_proxy.cc" ]
   sources += rebase_path(blink_core_sources_xml, "", "xml")
   sources +=
       rebase_path(blink_core_sources_xmlhttprequest, "", "xmlhttprequest")
@@ -475,6 +630,10 @@ component("core") {
       "Carbon.framework",
     ]
   }
+  visibility += brave_blink_renderer_core_visibility
+  public_deps += brave_blink_renderer_core_public_deps
+  sources += brave_blink_renderer_core_sources
+  deps += brave_blink_renderer_core_deps
 }

 # Certain sources should be compiled with heavy optimization enabled.
@@ -483,7 +642,19 @@ component("core") {
 blink_core_sources("core_hot") {
   sources = []
   sources += rebase_path(blink_core_hot_sources_dom, "", "dom")
+  sources -= rebase_path([
+                           "document.cc",
+                           "events/event_dispatcher.cc",
+                         ],
+                         "",
+                         "dom")
+  sources += [
+    "//brave/chromium_src/third_party/blink/renderer/core/dom/document.cc",
+    "//brave/chromium_src/third_party/blink/renderer/core/dom/events/event_dispatcher.cc",
+  ]
   sources += rebase_path(blink_core_hot_sources_html, "", "html")
+  sources -= rebase_path([ "parser/html_document_parser.cc" ], "", "html")
+  sources += [ "//brave/chromium_src/third_party/blink/renderer/core/html/parser/html_document_parser.cc" ]
   sources += rebase_path(blink_core_hot_sources_layout, "", "layout")
   sources += rebase_path(blink_core_hot_sources_css, "", "css")
